### DISCUSSION TITLE
Hosting multiple avalonia apps within a shell

### DISCUSSION THREAD

**Discussion by da-kil** [AvaloniaUI/Avalonia#14358] [Q&A]
I'm wondering if it is possible to host multiple avalonia apps within an avalonia app shell. 

I have a MainApplication with only a MainWindow. Let's say this MainWindow contains a menu with two buttons "Application 1" and "Application 2". If one clicks on "Application 1" button, application 1 starts (incl. bootstrap like filling ServiceCollection). If one clicks on "Application 2" button, application 1 is closing and application 2 starts.

![image](https://github.com/AvaloniaUI/Avalonia/assets/112584984/dd54d46b-1d8d-40aa-a68d-5104c06c784a)
MainWindow is just a shell with a menu.

After clicking on "Application 1" the app starts and renders within the shell as a kind of sub application.
![image](https://github.com/AvaloniaUI/Avalonia/assets/112584984/f4e50015-8794-4d28-8890-a01f37e293e8)

Is this even possible? If yes, how can I do it?

**Comment by maxkatz6** [comment]
Skipping question "Why?", answer is likely to be running multiple processes and hosting each window in the NativeControlHost. Unless somebody has a better idea.

**Comment by thevortexcloud** [comment]
I am not going to skip the why question, why do you need to do this? Surely this would be simpler and easier to manage as a single app?

**Reply by da-kil** [reply]
An application has its specific e.g. database (embedded) connection/file (using LiteDB). I want to inject the database into the data access service. For that I need to define the path at bootstrap time and register the database in the service collection. That is why I was trying this approach. In this case the application is like a project with its one database etc. If I close project 1 and open project 2 I kind of need the have a new service collection for that project. The easiest approach is to start a new application.
